<a href="https://colab.research.google.com/github/MarioViniciusBC/risco_credito/blob/main/pre_processamento_dados.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
!pip -q install plotly

In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px

### Base de dados - Risco de Crédito

In [4]:
# Carregamento da base de dados
dados = pd.read_csv('/content/credit_data.csv')

In [5]:
dados

,clientid,income,age,loan,default
0,1,66155.925095,59.017015,8106.532131,0
1,2,34415.153966,48.117153,6564.745018,0
2,3,57317.170063,63.108049,8020.953296,0
3,4,42709.534201,45.751972,6103.642260,0
4,5,66952.688845,18.584336,8770.099235,1
...,...,...,...,...,...
1995,1996,59221.044874,48.518179,1926.729397,0
1996,1997,69516.127573,23.162104,3503.176156,0
1997,1998,44311.449262,28.017167,5522.786693,1
1998,1999,43756.056605,63.971796,1622.722598,0


In [6]:
colunas = dados.columns
colunas

Index(['clientid', 'income', 'age', 'loan', 'default'], dtype='object')

Usamos o atributo `.columns` para acessar os nomes (ou rótulos) das colunas do DataFrame. Ele retorna um objeto nativo do próprio pandas chamado `Index`. Podemos converter para um ndarray usando `.to_numpy()`:

In [8]:
colunas_ndarray = colunas.to_numpy()
colunas_ndarray

array(['clientid', 'income', 'age', 'loan', 'default'], dtype=object)

In [7]:
numero_colunas = dados.shape[1]
numero_colunas

5

Com o método `.shape` verificamos que o número total de colunas é 5. Esse método retorna uma tupla no formato (número_de_linhas, número_de_colunas). Para pegar apenas as colunas, acessamos a posição `[1]`.

As variáveis são:
* **clientid** (id do  cliente) - Variável Categórica Nominal
* **income** (renda anual) - Variável Numérica Contínua
* **age** (idade) - Variável Numérica Contínua
* **loan** (dívida) - Variável Numérica Contínua
* **default** (indica se pagou ou não o empréstimo) - Variável Numérica Discreta

In [10]:
dados.head(10)

,clientid,income,age,loan,default
0,1,66155.925095,59.017015,8106.532131,0
1,2,34415.153966,48.117153,6564.745018,0
2,3,57317.170063,63.108049,8020.953296,0
3,4,42709.534201,45.751972,6103.642260,0
4,5,66952.688845,18.584336,8770.099235,1
5,6,24904.064140,57.471607,15.498598,0
6,7,48430.359613,26.809132,5722.581981,0
7,8,24500.141984,32.897548,2971.003310,1
8,9,40654.892537,55.496853,4755.825280,0
9,10,25075.872771,39.776378,1409.230371,0


In [11]:
dados.tail()

,clientid,income,age,loan,default
1995,1996,59221.044874,48.518179,1926.729397,0
1996,1997,69516.127573,23.162104,3503.176156,0
1997,1998,44311.449262,28.017167,5522.786693,1
1998,1999,43756.056605,63.971796,1622.722598,0
1999,2000,69436.579552,56.152617,7378.833599,0


In [12]:
dados.describe()

,clientid,income,age,loan,default
count,2000.000000,2000.000000,1997.000000,2000.000000,2000.000000
mean,1000.500000,45331.600018,40.807559,4444.369695,0.141500
std,577.494589,14326.327119,13.624469,3045.410024,0.348624
min,1.000000,20014.489470,-52.423280,1.377630,0.000000
25%,500.750000,32796.459717,28.990415,1939.708847,0.000000
50%,1000.500000,45789.117313,41.317159,3974.719419,0.000000
75%,1500.250000,57791.281668,52.587040,6432.410625,0.000000
max,2000.000000,69995.685578,63.971796,13766.051239,1.000000


A base de dados de crédito possui 2.000 registros e 5 atributos (`clientid`, `income`, `age`, `loan` e `default`). Uma análise estatística descritiva inicial revela dois problemas importantes na coluna `age` que exigirão tratamento na etapa de limpeza de dados: existem 3 valores ausentes (a contagem é de 1997, enquanto as outras colunas têm 2000) e a presença de idades negativas inconsistentes (valor mínimo de -52.42). Em relação ao perfil geral, os clientes possuem uma renda média de aproximadamente 45.331, pedem empréstimos com valor médio em torno de 4.444, e a taxa de inadimplência (`default`) da base é de 14,15%.